ovirt-engine-keycloak.spec.in

```diff
@@ -0,0 +1,113 @@
+%global __jar_repack 0
+
+%global product_name Keycloak SSO for oVirt Engine
+
+%global make_common_opts \\\
+	-j1 \\\
+	BUILD_VALIDATION=0 \\\
+	PACKAGE_NAME=%{name} \\\
+	RPM_VERSION=%{version} \\\
+	RPM_RELEASE=%{release} \\\
+	PREFIX=%{_prefix} \\\
+	SYSCONF_DIR=%{_sysconfdir} \\\
+	DATAROOT_DIR=%{_datadir} \\\
+	%{?EXTRA_BUILD_FLAGS:EXTRA_BUILD_FLAGS="%{EXTRA_BUILD_FLAGS}"}
+
+
+
+########################################################
+#  Keycloak overlay package
+########################################################
+Name:		ovirt-engine-keycloak
+Version:	@RPM_VERSION@
+Release:	@RPM_RELEASE@%{?dist}
+Summary:	%{product_name}
+Group:		Virtualization/Management
+License:	ASL 2.0
+URL:		http://keycloak.org
+BuildArch:	noarch
+Source:		%{name}-@RPM_VERSION@.tar.gz
+
+BuildRequires:	unzip
+
+Requires:	ovirt-engine-wildfly
+
+%description
+Keycloak SSO for oVirt Engine.
+
+########################################################
+#  Keycloak overlay setup package
+########################################################
+%package setup
+Summary:	%{product_name} setup
+Group:		Virtualization/Management
+
+BuildRequires:	python3
+BuildRequires:	python3-devel
+
+Requires:	ovirt-engine-setup-plugin-ovirt-engine-common >= 4.4.0
+Requires:	python%{python3_pkgversion}-ovirt-setup-lib
+Requires:	%{name} >= %{version}
```

Comment:
It's usually the other way around, at least for engine packages (which we versionlock, to prevent mismatch between code and data versions): The %{name} package Requires: %{name}-setup >= %{version}. This way, you can update setup packages with dnf update, then 'engine-setup' to update the app. No idea about current project - do you need to versionlock? Does it guarantee backward/forward compatibility?

Reply:
Keycloak overlay was intended to be extracted inside Wildfly installation and I am doing it differently. I put Keycloak files in its own location and symlink to wildfly whatever is necessary. Those symlinks are pretty static but I cannot exclude situation when something changes in layout in new version of Keycloak that would have to be reflected in *-setup package which means a version lock. 
I think your suggestion is valid and I like it more than my existing approach.

Reply:
GH issue created: #6